var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -470,6 +469,16 @@ def define_enable(suffix, value=None):
                                    'explicit_template_instantiation')
         ])
 
+        if spec.version >= Version('13'):
+            options.append(define_from_variant('CMAKE_CXX_STANDARD', 'cxxstd'))
+        else:
+            # Prior to version 13, Trilinos would erroneously inject
+            # '-std=c++11' regardless of CMAKE_CXX_STANDARD value
+            options.append(
+                define('Trilinos_CXX11_FLAGS',
+                       '-std=c++' + spec.variants['cxxstd'].value)
+            )
```

Comment:
```suggestion
            if spec.variants['cxxstd'].value == 11:
                stdflag = self.compiler.cxx11_flag
            else:
                stdflag = self.compiler.cxx14_flag
            if stdflag == '':
                stdflag = '-std=c++' + spec.variants['cxxstd'].value)
            options.append(define('Trilinos_CXX11_FLAGS', stdflag)
```
We can use compiler's c++ standard flag.
But gcc 6.0 or later's cxx14_flag is null string.

Reply:
Good point: we should be using the `spec.compiler.cxxNN_flag` property here: that'll automatically take care of the compiler-dependent issue. Assuming we don't add a c++98/03 variant, you can use:
```suggestion
            options.append(define(
                'Trilinos_CXX11_FLAGS',
                spec.compiler.cxx14_flag
                    if spec.variants['cxxstd'].value == '14'
                    else spec.compiler.cxx11_flag
            ))
```
(you may want to run a formatting check after trying this patch)

Reply:
This is a no-go with clang. With clang, I get:

```
==> [2022-01-30-10:44:48.936774] Error: AttributeError: 'CompilerSpec' object has no attribute 'cxx11_flag'

/Users/pakuber/test_spack_xyce/spack/var/spack/repos/builtin/packages/trilinos/package.py:481, in cmake_args:
        478                'Trilinos_CXX11_FLAGS',
        479                spec.compiler.cxx14_flag
        480                    if spec.variants['cxxstd'].value == '14'
  >>    481                    else spec.compiler.cxx11_flag
        482            ))
        483
        484        # ################## Trilinos Packages #####################
```